### DISCUSSION TITLE
Non-default locale canonical is wrong for dynamic pages due to different paths

### DISCUSSION THREAD

**Discussion by katerlouis** [nuxt-modules/i18n#1722] [Ideas]
We use storyblok as our cms and our pages use different paths per locale. Example:

`/datenschutz/` becomes `/en/privacy/`

But the resulting canonicals say `/datenschutz/` and `/en/datenschutz/`, which is a problem.

The pages coming from storyblok are not extended to the router for a very similar reason; `extendRoutes()` in nuxt.config cannot be told to have different paths per route (or can it?)

What do you guys suggest we should do? 

Thanks!

**Comment by katerlouis** [comment]
Maybe for the time being you guys can suggest a workaround to at least prevent the alternate canonicals to be generated?

**Reply by BobbieGoede** [reply]
It's been a long while (sorry), I'm slowly going through all issues (came across #1689).

This actually sounds more like an issue than a feature request, are you still experiencing this issue using the latest version? If so, could you provide a (minimal) reproduction demonstrating the problem you're running into?

**Comment by katerlouis** [comment]
@kazupon Thanks for reopening as a discussion. :)

**Comment by KartoffelCheetah** [comment]
I'm not sure as I do not use storyblok just nuxt2, but maybe this helps.
We use the head method to return the object generated by `this.$nuxtI18nHead`.
Instead of immediately returning the object generated we just modify this.
I personally just replaced the autogenerated canonical url with something else from a getter if that getter has a value.

```js
head () {
  const i18nHead = this.$nuxtI18nHead({ addSeoAttributes: true });

  return {
    ...i18nHead,
    link: [
      ...function * () {
        for (const lnk of i18nHead.link) {
          if (lnk.hid === 'i18n-can') {
            yield {
              ...lnk,
              href: this.$store.getters['seo/getCanonicalHref'] || lnk.href,
            };
          } else {
            yield lnk;
          }
        }
      }.bind(this)(),
    ],
  };
}
```
